markdown-cheat-sheet.md

```diff
@@ -1,121 +1,270 @@
+
+### Sample H3
+```
+
+will produce
+
+# Sample H1
+
+## Sample H2
+
+### Sample H3
+
+Headings introduce paragraph breaks automatically. However, you should still put a blank line before and a blank line after a heading.
+
+Note that there should only be one `H1` and it should be the first line in the markdown file. The `H1` style is typically used as a document title.
+
+Heading styles which need to move "upwards" to a higher number, should move one value at a time (`H1` to `H2`, for example, but not `H1` to `H3`). There are no constraints moving "downwards". `H6` is the biggest number (smallest sized heading).
+
+---
+
+## Horizontal Rules
+
+A horizontal rule is created using three dashes (`---`) on a line by itself and surrounded by blank lines.
+
+---
+
+## Coding - Block
+
+<pre>
+```bash
+declare ENV=${4:-sandbox}
+
+case "$1 $2 $3" in
 
-### Bold
+  'build manifests for')
+    if [[ ! $(git status | tee /dev/tty | grep 'nothing to commit, working tree clean') ]]; then
+```
+</pre>
+
+will produce
+
+```bash
+declare ENV=${4:-sandbox}
+
+case "$1 $2 $3" in
+
+  'build manifests for')
+    if [[ ! $(git status | tee /dev/tty | grep 'nothing to commit, working tree clean') ]]; then
+```
 
-Here is some **bold text**
+Note: You can specify alternative syntax highlighting based on the coding language, eg. `ruby`, `bash`, `php`, etc. by marking the *first set* of backticks as shown. If do not want syntax highlighting, use the \`\`\`text marker.
 
-### Italic
+Note: You should leave blank lines before the first and after the final markers.
 
-Here is some *italicized text*
+## Coding - In-line
 
-### Blockquote
+You can produce inline-code by wrapping the code fragment in single backticks:
 
-> blockquote
+```text
+This is some code: `echo something`
+```
 
-### Ordered List
+will produce
 
-1. First item
-2. Second item
-3. Third item
+This is some code: `echo something`
+
+---
 
-### Unordered List
+## Text Formatting
 
-- First item
-- Second item
-- Third item
+**Bold Text** is produced using `**Bold Text**`
 
-### Code
+*Italic Text* is produced using `*Italic Text*`
 
-`code`
+***Bold and Italic text*** is produced using `***Bold and Italic text***`.
 
-### Horizontal Rule
+~~Struck through text~~ is produced using `~~Struck through text~~`.
 
 ---
 
-### Link
+## Hyperlinks
 
-[Markdown Guide](https://www.markdownguide.org)
+GFMD will automatically detect URLs and convert them to links like this https://www.softwareone.com/. However, this also is a rule-breaking form. It's preferable to use the following:
 
-### Image
+```text
+This form is preferred: [Corporate Website](https://www.softwareone.com/)
+```
+
+This form is preferred: [Corporate Website](https://www.softwareone.com/)
+
+---
 
-![alt text](https://www.markdownguide.org/assets/images/tux.png)
+## Escape Sequence
 
-## Extended Syntax
+You can escape characters which would otherwise be actioned by preceding each actionable character with a backslash (`\`). For example \\\`
```

Comment:
Not quite sure what the example is trying to show - is it meant to show an escaped ` character?

Reply:
Maybe the example could be \`An Example\` - escaping the back tick characters?

Reply:
It also renders properly in GitHub. The escape sequence you're referring to renders as a backslash followed by a backtick.